WalletWasabi/WabiSabi/Models/DecompositionAlgs/Decomposition.cs

```diff
@@ -0,0 +1,59 @@
+using NBitcoin;
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace WalletWasabi.WabiSabi.Models.DecompositionAlgs
+{
+	public class Decomposition : IEnumerable<Coin>
+	{
+		private List<Coin> Coins { get; } = new();
+
+		public void Extend(Coin coin)
+		{
+			int index = Coins.BinarySearch(coin);
+			if (index < 0)
+			{
+				Coins.Insert(~index, coin);
+			}
+		}
+
+		public Money FaceValue()
+		{
+			return Coins.Sum(c => c.Amount);
+		}
+
+		public int OutputsVbytes()
+		{
+			return Coins.Sum(_ => Coin.OutputVbytes);
+		}
+
+		public decimal InputsVbytes()
```

Comment:
I'm guilty of this, Wasabi/NBitcoin uses Vsize and I keep using Vbytes, so that should probably match the wasabi convention

Reply:
Renamed